src/Kiota.Builder/CodeDOM/CodeComposedTypeBase.cs

```diff
@@ -71,4 +72,25 @@ protected override TChildType BaseClone<TChildType>(CodeTypeBase source, bool cl
         get;
         set;
     }
+    public bool IsComposedOfPrimitives() => Types.All(x => IsPrimitiveType(GetTypescriptTypeString(x, this)));
```

Comment:
the method that evaluates whether things are primitive or not should be a parameter for all those methods (Func)

Reply:
Completed